### DISCUSSION TITLE
Global install

### DISCUSSION THREAD

**Discussion by tripleo1** [sdkman/sdkman-cli#1073] [Ideas]
SDKMAN 5.14.1
Ubuntu 20.04

Let's say I wanted to install `node` for all users on my system

I would like to be able to say this:

```
sdk install jdk-whatever latest -R /opt
```

* The `-R` option is so `sdkman` will choose a directory name for me under `/opt`
* This also might need `sudo`
* And of course I put the version to `latest`. I don't know if this works currently or not.

**Comment by marc0der** [comment]
A few things to mention here. Firstly, we only support JVM SDKs so node/ruby/python will never be candidates on SDKMAN as they already have their own installers.

Regarding installing packages at arbitrary places on the filesystem: we purposefully do not allow this. SDKMAN installs it's candidates under the home directory on a per user basis. All candidates are always placed under the `.sdkman/candidates` folder. This is by design. That said, we allow alternative locations for the `.sdkman` folder as long as you have ownership of that location.

Last thing to mention is that we will never permit `sudo` privileges, for the sake of user security. SDKMAN is for the express purpose of downloading and installing SDKs for the currently active user and is not a package manager.

SDKMAN is an opinionated tool and has served its purpose very well up to this point. Our primary goal is to be a tool that makes the life of JVM developers easy. We provide you with an easy way to install all your tools so you can get on with the task of developing. The opinionated approach we have used has served us very well to achieve our goal over the years.

**Comment by tripleo1** [comment]
If you noticed I edited my message to install `java-whatever`, but only in the command line. I forgot that I mentioned it twice.

I actually built [fnm](https://github.com/Schniz/fnm), which means I had to download _another_ Rust nightly, but that's another story.

My issue was about sharing storage space between users, but I see your tool is very focused.

Thank you for writing it and thank you for your insight.

**Comment by tripleo1** [comment]
Also, I can't mark your comment as an answer.
